File: JavaScript/8js 1/calculateTip.js
/*
|--------------------------------------------------------------------------
| Tip Calculator
|--------------------------------------------------------------------------
|
*/

function calculateTip(amount, rating) {
   switch (rating.toLowerCase()) {
      case "terrible":
         return 0;
      case "poor":
         return Math.ceil(amount * 0.05);
      case "good":
         return Math.ceil(amount * 0.1);
      case "great":
         return Math.ceil(amount * 0.15);
      case "excellent":
         return Math.ceil(amount * 0.2);
      default:
         return "Rating not recognised";
   }
}


let calculateTip = (a, r) => {
   const i = ["terrible", "poor", "good", "great", "excellent"].indexOf(r.toLowerCase());
   return i === -1 ? "Rating not recognised" : Math.ceil(i * a * 0.05);
};


function calculateTip(amount, rating) {
   var tips = {
      terrible: 0,
      poor: 0.05,
      good: 0.1,
      great: 0.15,
      excellent: 0.2,
   };
   rating = rating.toLowerCase();
   return rating in tips ? Math.ceil(amount * tips[rating]) : "Rating not recognised";
}


console.log(calculateTip(20, "Excellent"), 4);
console.log(calculateTip(26.95, "good"), 3);

File: Python/8py 1/calculate_tip.py
"""
|--------------------------------------------------------------------------
| Tip Calculator
|--------------------------------------------------------------------------
|
"""

from math import ceil
def calculate_tip(amount, rating):
   tips = {'terrible': 0,'poor' : .05,'good' : .1,'great' : .15,'excellent' : .2}
   return ceil(amount * tips[rating.lower()]) if rating.lower() in tips else 'Rating not recognised'
   

def calculate_tip(a, r):
   d = dict(zip(['terrible', 'poor','good','great','excellent'], [0, 5, 10, 15, 20]))
   r = r.lower()
   if r in d:
      return (ceil(d[r]*a/100))
   else:
      return 'Rating not recognised'
   

def calculate_tip(amount, rating):
   ratings = {
      "Terrible": 0,
      "Poor": .05,
      "Good": .10,
      "Great": .15,
      "Excellent":.20
   }
   try:
      return ceil(amount * ratings.get(rating.title()))
   except:
      return "Rating not recognised"
   

print(calculate_tip(30, "poor"), 2)
print(calculate_tip(20, "Excellent"), 4)
print(calculate_tip(20, "hi"), 'Rating not recognised')
print(calculate_tip(107.65, "GReat"), 17)
print(calculate_tip(20, "great!"), 'Rating not recognised')